app/services/holdings_request_service.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+class HoldingsRequestService
+  attr_reader :args, :locations, :libraries, :exclude
+
+  def initialize(args)
+    @args = args
+    @libraries_data = libraries_data
+    @locations = libraries_data['locations']
+    @libraries = libraries_data['libraries']
+    @exclude = libraries_data['exclude_from_email_availability']
+  end
+
+  def make_holdings_request
+    sirsi_uri = URI(Settings.sirsi_url + args)
+
+    Net::HTTP.get(sirsi_uri)
+  end
+
+  def holdings_availability
+    holdings = Hash.from_xml(make_holdings_request)
+    availability = get_location_data(holdings)
+    if availability.present?
+      availability_to_string(availability)
+    end
+  end
+
+  private
+
+    def get_location_data(holdings)
+      call_info = holdings['LookupTitleInfoResponse']['TitleInfo']['CallInfo']
+
+      if call_info.is_a?(Array)
+        get_data_from_call_info(call_info)
+      else
+        item_location_data = call_info['libraryID'] == 'ONLINE' ? [] : get_item_location_data(call_info)
+        if item_location_data.present?
+          [{ 'library' => libraries[call_info['libraryID']],
+             'call_number' => call_info['callNumber'],
+             'items' => item_location_data }]
+        end
+      end
+    end
+
+    def get_data_from_call_info(call_info)
+      availability = []
+      call_info.each do |holding|
+        if availability.size >= 30
+          @view_more = true
+          break
+        end
+        item_location_data = holding['libraryID'] == 'ONLINE' ? [] : get_item_location_data(holding)
```

Comment:
same here, they seem to be doing similar job. depending on where you call that method, either call_info or holding could be passed in as an argument.